pkg/store/downtime_store.go

```diff
@@ -0,0 +1,360 @@
+package store
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
+	"github.com/vshn/vshn-sli-reporting/pkg/lieutenant"
+	"github.com/vshn/vshn-sli-reporting/pkg/types"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+type dbDowntimeWindow struct {
+	ID           string `db:"id"`
+	StartTime    int64  `db:"start_time"`
+	EndTime      int64  `db:"end_time"`
+	Title        string `db:"title"`
+	Description  string `db:"description"`
+	ExternalID   string `db:"external_id"`
+	ExternalLink string `db:"external_link"`
+	Affects      string `db:"affects"`
+}
+
+type downtimeStoreImpl struct {
+	db         *sqlx.DB
+	lieutenant lieutenant.Client
+}
+
+type DowntimeStore interface {
```

Comment:
Arguable the same here: I personally would move the interface to the package using it.

Reply:
Multiple users of the interface on the receiver side? It really depends. If both are completely unrelated i'd copy. If both are related I'd organise the packages like this e.g. `thing/server, thing/backgroundworker` and define the interface in `thing`. If multiple is a package and its test/mock I'd just depend on the original package.

In general I see go interfaces more like duck typing from dynamic languages but more explicit.